mlflow/pipelines/utils/execution.py

```diff
@@ -400,6 +400,25 @@ def _create_makefile(pipeline_root_path, execution_directory_path, template) ->
 
     if template == "regression/v1":
         makefile_to_use = _REGRESSION_MAKEFILE_FORMAT_STRING
+        steps_folder_path = os.path.join(pipeline_root_path, "steps")
+        if not os.path.exists(steps_folder_path):
+            os.mkdir(steps_folder_path)
+        for required_file in [
+            "ingest.py",
+            "split.py",
+            "train.py",
+            "transform.py",
+            "custom_metrics.py",
+        ]:
+            required_file_path = os.path.join(steps_folder_path, required_file)
+            if not os.path.exists(required_file_path):
+                with open(required_file_path, "w") as _:
+                    pass
+            if not os.path.exists(required_file_path):
+                raise ValueError(
+                    f"Can not find required file {required_file_path} from steps folder. "
+                    "Please create empty python file if the step is not used."
+                )
```

Comment:
Do we need this block? Looks like `not os.path.exists(required_file_path)` always returns False as we create an empty file at `required_file_path` in line 415.

Reply:
It's probably not required. Any chance the `open(required_file_path, "w")` fails to create the file?

Reply:
There can be an IO error, consider:
```
try:
    open(required_file_path, 'w').close()
except OSError e:
    # Do something with e
```
If you want to add context to the error.